src/Elastic.Apm/Filters/TransactionIgnoreUrlsFilter.cs

```diff
@@ -15,8 +16,20 @@ namespace Elastic.Apm.Filters
 	internal class TransactionIgnoreUrlsFilter
 	{
 		private readonly IConfigSnapshot _configSnapshot;
+
 		public TransactionIgnoreUrlsFilter(IConfigSnapshot configSnapshot) => _configSnapshot = configSnapshot;
-		public ITransaction Filter(ITransaction transaction) =>
-			WildcardMatcher.IsAnyMatch(_configSnapshot.TransactionIgnoreUrls, transaction?.Context?.Request?.Url?.PathName) ? null : transaction;
+
+		public ITransaction Filter(ITransaction transaction)
+		{
+			if (transaction is Transaction realTransaction)
+			{
+				// If there is no context, there is no URL either, therefore this transaction can't be filtered based on the URL.
+				if (!realTransaction.IsContextCreated)
```

Comment:
This isn't strictly related, but now that we register the same filters (including this one) in the mock payload senders in our tests as we do in `PayloadSenderV2` in prod, turns out that this filter created `transaction.Context` always, even if it wasn't needed.  ...so now we check for that.